authentication/openshift.go

```diff
@@ -501,7 +504,36 @@ func (a OpenShiftAuthenticator) Middleware() Middleware {
 
 func (a OpenShiftAuthenticator) GRPCMiddleware() grpc.StreamServerInterceptor {
 	return grpc_middleware_auth.StreamServerInterceptor(func(ctx context.Context) (context.Context, error) {
-		return ctx, status.Error(codes.Unimplemented, "internal error")
+		httpReq := http.Request{Header: map[string][]string{}}
+
+		// set authorization header
+		md, ok := metadata.FromIncomingContext(ctx)
+		if !ok {
+			return ctx, status.Error(codes.Internal, "metadata error")
+		}
+		for _, val := range md.Get("Authorization") {
+			httpReq.Header.Add("Authorization", val)
+		}
+
+		// set peer certificates
+		peerContext, ok := peer.FromContext(ctx)
+		if ok {
+			tlsInfo := peerContext.AuthInfo.(credentials.TLSInfo)
+			httpReq.TLS = &tls.ConnectionState{
+				PeerCertificates: tlsInfo.State.PeerCertificates,
+			}
+		}
```

Comment:
This sets the PeerCertificats for the mTLS case right?

Reply:
yes